Mathlib/AlgebraicGeometry/Morphisms/IsIso.lean

```diff
@@ -0,0 +1,45 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.AlgebraicGeometry.Morphisms.OpenImmersion
+import Mathlib.Topology.IsLocalHomeomorph
+
+/-!
+
+# Being an isomorphism is local at the target
+
+-/
+
+open CategoryTheory MorphismProperty
+
+namespace AlgebraicGeometry
+
+lemma isomorphisms_eq_isOpenImmersion_inf_surjective :
+    isomorphisms Scheme = (@IsOpenImmersion ⊓ @Surjective : MorphismProperty Scheme) := by
+  ext
+  exact (isIso_iff_isOpenImmersion _).trans
+    (and_congr Iff.rfl ((TopCat.epi_iff_surjective _).trans (surjective_iff _).symm))
+
+lemma isomorphisms_eq_stalkwise :
+    isomorphisms Scheme = (isomorphisms TopCat).inverseImage Scheme.forgetToTop ⊓
+      stalkwise (fun f ↦ Function.Bijective f) := by
+  rw [isomorphisms_eq_isOpenImmersion_inf_surjective, isOpenImmersion_eq_inf,
```

Comment:
I think we should have `IsHomeomorphism` and then state this in terms of it, i.e. `isomorphisms Scheme = topologically IsHomeomorphism ⊓ stalkwise (fun f ↦ Function.Bijective f)`.

Reply:
Hmmm I think `isomorphisms` is exactly this and already has some useful API around it.

Reply:
But in terms of category theory which does not always play nicely with the type theoretic topology library.

Reply:
There is #12533 but it seems like a stale PR.

Reply:
Hm that's sad. Okay, let's merge this one as is and refactor later (or just add the suggested formulation) when we have `IsHomeomorphism`.

Reply:
I've asked on the relevant Zulip thread whether the author intend to continue and would probably take over if the author has decided to abandon it.
I'll add the suggestion once that PR is merged. I think this lemma is still useful though and I'll probably still keep this.